File: src/lyric/router.ts
import { Router } from 'express';

import {
  handleGetLyrics,
  handleGetLyric,
  handleCreateLyric,
  handleUpdateLyric,
  handleDeleteLyric,
} from './controller';
import authMiddleware from '../auth/middleware';

const lyricRouter = Router();

lyricRouter.get('/', handleGetLyrics);
lyricRouter.get('/:id', handleGetLyric);
lyricRouter.post('/', authMiddleware, handleCreateLyric);
lyricRouter.put('/:id', authMiddleware, handleUpdateLyric);
lyricRouter.delete('/:id', authMiddleware, handleDeleteLyric);

export default lyricRouter;

File: lyrics/router.go
package lyrics

import (
	"github.com/AndreyGalchevski/strident-api/auth"
	"github.com/gorilla/mux"
)

func InitLyricsRouter(r *mux.Router) {
	r.HandleFunc("/lyrics", handleGetLyrics).Methods("GET")

	r.HandleFunc("/lyrics/{id}", auth.VerifyAuthorization(handleGetLyricByID)).Methods("GET")
	r.HandleFunc("/lyrics", auth.VerifyAuthorization(handlePostLyric)).Methods("POST")
	r.HandleFunc("/lyrics/{id}", auth.VerifyAuthorization(handlePatchLyric)).Methods("PATCH")
	r.HandleFunc("/lyrics/{id}", auth.VerifyAuthorization(handleDeleteLyric)).Methods("DELETE")
}
